public/pages/AnomalyCharts/containers/AnomaliesChart.tsx

```diff
@@ -99,85 +68,29 @@ interface AnomaliesChartProps {
-  const [anomalySummary, setAnomalySummary] = useState<AnomalySummary>(
-    INITIAL_ANOMALY_SUMMARY
-  );
-  const [showAlertsFlyout, setShowAlertsFlyout] = useState<boolean>(false);
-  const [alertAnnotations, setAlertAnnotations] = useState<any[]>([]);
-  const [isLoadingAlerts, setIsLoadingAlerts] = useState<boolean>(false);
-  const [totalAlerts, setTotalAlerts] = useState<number | undefined>(undefined);
-  const [alerts, setAlerts] = useState<MonitorAlert[]>([]);
-  const [zoomRange, setZoomRange] = useState<DateRange>({
-    ...props.dateRange,
-  });
-  const [zoomedAnomalies, setZoomedAnomalies] = useState<any[]>([]);
 
+export const AnomaliesChart = React.memo((props: AnomaliesChartProps) => {
   const [datePickerRange, setDatePickerRange] = useState({
     start: 'now-7d',
     end: 'now',
   });
 
-  useEffect(() => {
-    const anomalies = prepareDataForChart(props.anomalies, zoomRange);
-    setZoomedAnomalies(anomalies);
-    setAnomalySummary(
-      !props.bucketizedAnomalies
-        ? getAnomalySummary(
-            filterWithDateRange(props.anomalies, zoomRange, 'plotTime')
-          )
-        : props.anomalySummary
-    );
-    setTotalAlerts(filterWithDateRange(alerts, zoomRange, 'startTime').length);
-  }, [props.anomalies, zoomRange]);
-
-  const handleZoomRangeChange = (start: number, end: number) => {
-    setZoomRange({
-      startDate: start,
-      endDate: end,
-    });
-    props.onZoomRangeChange(start, end);
-  };
-
-  useEffect(() => {
-    async function getMonitorAlerts(monitorId: string, startDateTime: number, endDateTime: number) {
-      try {
-        setIsLoadingAlerts(true);
-
-        const result = await dispatch(
-          searchAlerts(monitorId, startDateTime, endDateTime)
-        );
-
-        setIsLoadingAlerts(false);
-        setTotalAlerts(get(result, 'data.response.totalAlerts'))
-
-        const monitorAlerts = convertAlerts(result);
-        setAlerts(monitorAlerts);
-        const annotations = generateAlertAnnotations(monitorAlerts);
-        setAlertAnnotations(annotations);
-      } catch (err) {
-        console.error(`Failed to get alerts for monitor ${monitorId}`, err);
-        setIsLoadingAlerts(false);
-      }
-    }
-    if (props.monitor && props.dateRange.startDate) {
-      getMonitorAlerts(props.monitor.id, props.dateRange.startDate, props.dateRange.endDate);
-    }
-  }, [props.monitor, props.dateRange.startDate]);
-
-  const anomalyChartTimeFormatter = niceTimeFormatter([
-    zoomRange.startDate,
-    zoomRange.endDate,
-  ]);
+  const anomalies = get(props.anomaliesResult, 'anomalies', []);
 
   const handleDateRangeChange = (startDate: number, endDate: number) => {
     props.onDateRangeChange(startDate, endDate);
-    handleZoomRangeChange(startDate, endDate);
+    props.onZoomRangeChange(startDate, endDate);
   };
 
-  const showLoader = useDelayedLoader(props.isLoading || isLoadingAlerts);
+  const showLoader = useDelayedLoader(props.isLoading);
```

Comment:
Any impact to single entity detector by removing `isLoadingAlerts `?

Reply:
no. it will be there for `AnomalyDetailsChart`